SIPS/sip-keeper-system.md

```diff
@@ -0,0 +1,163 @@
+
+This would allow price updates to be less than 0.5% / 1.0% for Synthetix exchanges and futures that require faster updates and less slippage when there is a large notional value of settlements / exchanges that are waiting for the next price update, keepers can request an update from the Chainlink oracles.
+
+For example, if there is a $`1m exchange into sXAU` that is waiting for settlement (next price update), rather than waiting for the next Chainlink price update when a 1.0% deviation is detected, the keeper could request for a price update earlier.
+
+The current Chainlink oracles work on a 0.5% / 1.0% deviation due to practical limitations such as block time, gas costs of multiple nodes publishing prices on chain and the cost of LINK required to pay for smaller deviations. Covering the costs of pulling a next price update on chain would allow for a quicker price updates. The price update can also be consumed by other consumers of the Chainlink oracles as per normal update.
+
+### Resolving Binary option markets
+
+Binary options would allow market resolution and market expiry maintenance to be executed by keepers to help maintain the binary options markets.
+
+- `BinaryOptionMarketManager.resolveMarket()`
+- `BinaryOptionMarketManager.expireMarkets()`
+
+### Executing and Claiming Virtual Tokens
+
+Virtual Tokens are a proposed mechanism for representing ownership of a synth exchange that is waiting for the fee relcamation to be settled and swapped.
+
+Virtual tokens representing a claim on un-settled synths after an exchange would benefit from keepers executing them and claiming them for the end user.
+
+This would improve the UX of users who trade with virtual tokens such that if they hold vDAI after exchanging from `sETH -> sUSD -> vDAI`, the keeper will execute the `vDAI` exchange after the fee relcamation waiting period into DAI directly.
+
+For the user the UX of finalising the redemption of the `vDAI` into `DAI` would be seamless as it will be completed by keepers and the gas paid for them. The execution of the exchange would be happen after the oracle update waiting period is expired.
+
+### Synthetix Futures keepers and liquidation
+
+Keepers will play an important role in managing the Synthetix Futures system. The keepers will be incentivised to perform functions as required to maintain trader's positions and fee reclamation settlement for these positions.
+
+**Liquidations**
+
+A `Liquidation Fee` is paid to keepers, from the trader's maintenance margin, as a strong incentive to liquidate positions that have been liquidated. Synthetix futures positions that have gone below their liquidation price / maintenance margin would require keepers to liquidate them in a timely manner.
+
+The keeper system wouldn't be required to pay a reward to cover the cost of liquidating futures position.
+
+# Rationale
```

Comment:
empty?

Reply:
@justinjmoses  Have updated this section now